src/bosh-director/lib/cloud/external_cpi.rb

```diff
@@ -1,6 +1,28 @@
 require 'membrane'
 require 'open3'
 
+# Robbed from:
+# https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/io.rb
+class IO
+  def readline_nonblock(sep = $INPUT_RECORD_SEPARATOR)
+    line = +""
+    buffer = +""
+
+    loop do
+      break if buffer == sep
+
+      read_nonblock(1, buffer)
+      line.concat(buffer)
+    end
+
+    line.freeze
+  rescue IO::WaitReadable, EOFError => e
```

Comment:
Should we `rescue` also `IO::WaitReadable` in https://github.com/cloudfoundry/bosh/blob/f4ae87f7e2ff696ad9048d2a58bdcd6b9a2eee8e/src/bosh-director/lib/cloud/external_cpi.rb#L121 or did we leave it out on purpose?

Reply:
@rkoster `bundle exec rspec spec/unit/cloud/external_cpi_spec.rb:648` from `src/bosh-director` folder worked for me.

Reply:
@beyhan good catch. Looks like we should rescue IO::WaitReadable and do nothing with it so that IO.select can retry the read, as described here: https://github.com/ruby/ruby/blob/98cc6d158b501a4c30aa578239b35085a4612b94/io.rb#L36

Reply:
running from `src/bosh-director` did the trick, was able to confirm the unit test indeed fails when reverting the nonblocking change (and pass when applied).